[RPC] Use qualified name str directly in RPC torch script code path

xush6528 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34733 [RPC] Use qualified name str directly in RPC torch script code path**

simplify

Differential Revision: [D20442573](https://our.internmc.facebook.com/intern/diff/D20442573/)

xush6528 committed:
[RPC] Use qualified name str directly in RPC torch script code path

simplify

Differential Revision: [D20442573](https://our.internmc.facebook.com/intern/diff/D20442573/)

[ghstack-poisoned]

zhaojuanmao commented on torch/csrc/distributed/rpc/init.cpp:
```diff
@@ -421,17 +421,16 @@ If the future completes with an error, an exception is thrown.
-         const py::object& userCallable,
+         const std::string& qualifiedNameStr,
          const py::tuple& argsTuple,
          const py::dict& kwargsDict) {
```
nit: change them back to py::args and py::kwargs?

xush6528 replied:
Changing this will cause memory leak. I will re-land the rest and debug on this later.

mrshenli reviewed (commented):
Agree with @zhaojuanmao  on arg type consistency. Others LGTM!

Could you please rebase, test failures should have been fixed.

xush6528 committed:
Update on "[RPC] Use qualified name str directly in RPC torch script code path"


simplify

Differential Revision: [D20442573](https://our.internmc.facebook.com/intern/diff/D20442573/)

[ghstack-poisoned]

xush6528 committed:
Update on "[RPC] Use qualified name str directly in RPC torch script code path"


simplify

Differential Revision: [D20442573](https://our.internmc.facebook.com/intern/diff/D20442573/)

[ghstack-poisoned]

xush6528 committed:
Update on "[RPC] Use qualified name str directly in RPC torch script code path"


simplify

Differential Revision: [D20442573](https://our.internmc.facebook.com/intern/diff/D20442573/)

[ghstack-poisoned]

mrshenli reviewed (approved):
Thanks for adding this!
